UKHomeOffice/nmsw-ui
# NMSW-1306 Refactor CYA page to use new declaration endpoint

# Ticket



----

## AC

- update page to use /declaration instead of /declaration/attachments endpoint for getting declaration information
- refactor AMG checking statuses as the `pending` items of the API response are now removed and replaced with more detailed status counts

----

## To test

Draft report
- create a declaration with FAL1 only
- type in the url for CYA using a draft reports declaration ID
- try to submit report
- > you should see errors for passenger and crew files
- 
Submitted report
- check page still renders
- check FAL1, 5, 6, supporting documents are still downloadable
- check General Declaration information still displays
- check you can still cancel the report

----

## Notes

Updated all the fixture files to match the new responses and tried to consolidate them
